spec/javascript/controllers/gtm_controller_spec.js

```diff
@@ -0,0 +1,67 @@
+import { Application } from 'stimulus' ;
+import GtmController from 'gtm_controller' ;
+
+describe('GtmController', () => {
+  document.head.innerHTML = `<script></script>` ;
+  document.body.innerHTML = `
+  <div data-controller="gtm" data-gtm-action="test" data-gtm-event="test"></div>
+  ` ;
+
+  // window appears to not be getting redefined between runs, so remove manually
+  afterEach(() => { delete window.gtag }) ;
+
+  describe("with cookie already set", () => {
+    beforeEach(() => {
+      Object.defineProperty(window.document, 'cookie', {
+        writable: true,
+        value: 'GiTBetaCookie=Accepted; expires=Fri, 31 Dec 2021 12:00:00 UTC'
```

Comment:
This test will start failing on that date, we could always create a `Date` in the future in place of this hard coded string

Reply:
Yeh, good call - took this from one of the existing controller tests without thinking about it.